pandas/core/sparse/frame.py

```diff
@@ -73,6 +75,9 @@ def __init__(self, data=None, index=None, columns=None, default_kind=None,
             if columns is None:
                 raise Exception("cannot pass a series w/o a name or columns")
             data = {columns[0]: data}
+        elif isinstance(data, BlockManager):
+            if default_fill_value is None:
+                default_fill_value, _ = Counter([b.fill_value for b in data.blocks]).most_common(1)[0]
```

Comment:
I don't know if this is kosher or not... 

Basically I kept running into the issue of

If you create a SparseDataFrame with a bunch of SparseSeries / SparseArrays that have fill_values like == 1 or ==0 or something then it doesn't persist that to default_fill_value. That seems like an enhancement and I could take it out of this PR but it helped me test.

Reply:
This seems like a recipe for surprise. I'd hate for `df.reindex()` to do something different, based on the types of blocks that `SparseDataFrame` happened to be initialized with. If a user explicitly sets `default_fill_value` that's one thing, but inferring it from the data seems problematic.

Reply:
I can take it out it didn't add much tbh. was just making my testing easier ;)